### PR TITLE
minor changes to make method lookup diagnostic code easier to read

### PR DESCRIPTION
**Pull request by BoxyUwU**
The end result of around 4 days of trying to understand this 1000+ line long function- a bunch of tiny nitpicks

r? @compiler-errors

### CODE REVIEW

File: compiler/rustc_hir_typeck/src/method/suggest.rs [hunk 1]
```diff
@@ -620,22 +611,6 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         match self.tcx.hir().get_if_local(impl_def_id) {
                             // Unmet obligation comes from a `derive` macro, point at it once to
                             // avoid multiple span labels pointing at the same place.
-                            Some(Node::Item(hir::Item {
```
Comment (compiler-errors) [inline compiler/rustc_hir_typeck/src/method/suggest.rs:623]: I'm pretty convinced that this code will never be triggered, but if it is, someone will report it as an ICE :sweat_smile:

Comment (compiler-errors) [reply compiler/rustc_hir_typeck/src/method/suggest.rs:623]: For this to be triggered, we need to have a nested obligation of a trait be reported as an unsatisfied predicate for a method to be applicable. We only construct nested trait obligations with `ImplDerivedObligation` causes for auto traits, which have no methods.

Comment (chenyukang) [reply compiler/rustc_hir_typeck/src/method/suggest.rs:623]: #105732 is here,
maybe we should add an `delay_span_bug` here for match with `Some(node)`?
@compiler-errors

File: compiler/rustc_hir_typeck/src/method/suggest.rs [hunk 2]
```diff
@@ -248,7 +248,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         match error {
             MethodError::NoMatch(NoMatchData {
```
Comment (chenyukang) [inline compiler/rustc_hir_typeck/src/method/suggest.rs:250]: This block is tooo long, about 800 lines of code!
Is it worth to extract to a seperate sub-methods?
@compiler-errors

**Pull request merged**